api/client/src/main/java/org/projectnessie/client/http/HttpClientBuilderImpl.java

```diff
@@ -240,32 +252,56 @@ public HttpClient build() {
             .addAllResponseFilters(responseFilters)
             .isHttp11Only(!http2Upgrade)
             .followRedirects(followRedirects)
-            .forceUrlConnectionClient(forceUrlConnectionClient)
             .authentication(authentication)
             .build();
 
-    return ImplSwitch.FACTORY.apply(config);
+    String clientName = httpClientName;
+    if ("HTTP".equalsIgnoreCase(clientName)) {
```

Comment:
Is this special name really worth it? What's the added value compared to simply not providing any client name?

Reply:
Why not just put the preferred impl into the `IMPLEMENTATIONS` map under the `HTTP` key?

Reply:
`HTTP` refers to `org.projectnessie.client.http.NessieHttpClientBuilderImpl.name()`

Reply:
true, but we resolve it to the "preferred" client, right? My point is: why have complex logic involving `null` for that when we can put the preferred impl. under `http` in the map?

Reply:
Need the `null` & `forceUrlConnectionClient` handling anyway

Reply:
I don't understand. `HTTP` is the name of the `NessieHttpClientBuilderImpl` - it is NOT an explicitly documented discriminator - it has to be handled as if no client name has been set.

Reply:
but, I think `HTTP` is a client discriminator that we supported before, right?

In any case, it can be provided by a user. Also `null` can be provided by the user. I'm suggesting to convert `null` to `HTTP` and hopefully simplify the following look up logic (basically avoid special cases and always get the impl. from the map). WDYT?